src/platform/plugins/shared/dashboard/public/dashboard_app/url/bwc/extract_control_group_state.ts

```diff
@@ -7,95 +7,146 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import type { ControlWidth } from '@kbn/controls-schemas';
-import type { DashboardState, DashboardControlsState } from '../../../../common';
+import { get } from 'lodash';
+import type { ControlsGroupState } from '@kbn/controls-schemas';
+import {
+  DEFAULT_AUTO_APPLY_SELECTIONS,
+  DEFAULT_IGNORE_VALIDATIONS,
+  DEFAULT_USE_GLOBAL_FILTERS,
+} from '@kbn/controls-constants';
+import type { DashboardState } from '../../../../common';
 
-interface State816To819 {
-  controlGroupState: {
-    initialChildControlState: Record<
-      string,
-      object & {
-        grow: boolean;
-        order: number;
-        type: string;
-        width: ControlWidth;
-      }
-    >;
-  };
-}
-
-const isState816To819 = (state: unknown): state is State816To819 =>
-  Boolean((state as State816To819).controlGroupState?.initialChildControlState);
+export function extractControlGroupState(state: { [key: string]: unknown }): {
```

Comment:
This file really shows how many breaking changes we've made to the URLs over the years. Thanks for untangling all of this and ensuring we aren't repeating past mistakes with Controls Anywhere.